### PR TITLE
Muon multiple fitting gui

### LINKED ISSUE #19149: Muon Analysis fitting with shared parameters GUI interface
The current version of Mantid (3.9.1) can perform fits to multiple data set with shared parameters. However, the layout of the GUI is very different from the single fit muon interface. As a result users have found the interface unintuitive and difficult to navigate. 

This piece of work will redesign the GUI for multiple fits, so that it more closely resemble the single fit interface and is more intuitive. 

The output will be a new GUI interface for the muon interface when fitting to multiple data sets.

### PR DESCRIPTION
**Pull request by AnthonyLim23**
This is the first phase of an upgrade to the multiple fitting GUI in muon analysis to make it more user friendly. 

If a single fit (one period and one group/pair) is done then the labels are wrong and already has an issue #19544 

**To test:**

There are multiple tests. I recommend using HIFI data (never use the current run) as it has both multiple periods and groups. 

0. The input parameters change after a simultaneous fit if a different data set is selected for display (assuming it is not global). 

1. Changing the Groups/Pairs to fit option updates the GUI correctly. So the relevant data should be listed in the Display parameters for drop down menu. A fit should use the relevant data (can check by doing a tiled plot of the output workspace). 

2. Changing the periods to fit option updates the GUI correctly. The response should be similar to the above. 

3. If custom is selected from the drop down menus (in 1 and 2) then the relevant pop up is produced and data is passed to the GUI correctly. If you press the Reselect Groups/Pairs or Periods buttons (will only be enabled if the relevant drop down menu is set to custom) the pop up menu should reappear. 

4. If the data has multiple periods, the Combine Periods button will produce a pop up that allows users to create combinations of periods. This should be added to the period drop down and pop ups. Check that the data is passed to the GUI correctly. 

5. Changing the start and end alters the fit. 

6. User can switch between multi and single fit options and the fits work (numbers don't matter).

7. Single fit produces the same answers as before. If you change the workspace in data analysis there is a bug #19571 

Fixes #19149  and #19544 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
I have updated the comparison branch, to remove the files from the other PR

**Comment by AnthonyLim23** [comment]
Just found a bug, which I am going to fix now

**Comment by DTasev** [comment]
Issues found from testing:
- `Combine Periods`, `Periods` and `Groups/Pair` buttons in multiple fitting stop Mantid from closing, if you try closing Mantid while one of them is open
  - You can also open multiple instances of each, maybe the view wasn't set as the parent? (I haven't looked at the code yet)
- The Global checkboxes in the `Fit Function` section of multiple fitting are grayed out and look like inactive, yet they seem to be active and clickable, I think they should be reverted to look like normal enabled checkboxes (different issue ?)

Apart from that, after following the testing instructions and with some help from @AnthonyLim23 I went through the testing directions and found no other issues, everything seemed to be working as described.

**Comment by AnthonyLim23** [comment]
I can add a check for if the pop-ups are open. I'm not sure about the closing. 

The global checkboxes are unchanged from the current release.

File: MantidQt/CustomInterfaces/src/Muon/MuonAnalysis.cpp [hunk 1]
```diff
@@ -318,6 +318,18 @@ void MuonAnalysis::initLayout() {
   // Manage User Directories
   connect(m_uiForm.manageDirectoriesBtn, SIGNAL(clicked()), this,
           SLOT(openDirectoryDialog()));
+  // m_uiForm.
```
Comment (DTasev) [inline MantidQt/CustomInterfaces/src/Muon/MuonAnalysis.cpp]: unfinished comment?

File: MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/MuonFitPropertyBrowser.h [hunk 2]
```diff
@@ -149,6 +190,32 @@ private slots:
   std::string m_simultaneousLabel;
   QtProperty *m_normalization;
   mutable QStringList m_normalizationValue;
```
Comment (DTasev) [inline MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/MuonFitPropertyBrowser.h]: Is there a reason for these to be `mutable`? They're never changed inside a `const` function

**Review by dtasev — CHANGES_REQUESTED** [review]
Just a few minor comments

File: MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/MuonFitPropertyBrowser.h (hunk 2 above)
Comment (AnthonyLim23) [inline MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/MuonFitPropertyBrowser.h]: for which one to be mutable?

m_normalizationValue is overwritten

File: MantidQt/CustomInterfaces/src/Muon/MuonAnalysis.cpp (hunk 1 above)
Comment (AnthonyLim23) [inline MantidQt/CustomInterfaces/src/Muon/MuonAnalysis.cpp]: thought I had caught all of them. Its an old bit of code, not needed anymore (as you have seen). 

Thank you for spotting it

File: MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/MuonFitPropertyBrowser.h (hunk 2 above)
Comment (DTasev) [inline MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/MuonFitPropertyBrowser.h]: `m_normalizationValue` and a few of the other variables below it are set to mutable

Comment (AnthonyLim23) [inline MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/MuonFitPropertyBrowser.h]: these are all overwritten at other points in the code. 

m_normalization -> in TFAsymmfit
m_groupsToFitOptions -> in addGroupCheckbox
m_ShowGroupValues -> in updateGroupDisplay

similarly for periods.

Comment (DTasev) [inline MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/MuonFitPropertyBrowser.h]: They're not changed inside `const` functions, the `mutable` keyword does not do anything here, it's only confusing

Comment (AnthonyLim23) [inline MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/MuonFitPropertyBrowser.h]: I will try removing it. I have a vague recollection of it not working.

**Comment by AnthonyLim23** [comment]
fixed them

**Review by dtasev — APPROVED** [review]

**Comment by DTasev** [comment]
Everything looks good, I did the testing after the review comments and other fixes have been added, and found no further issues. I'm happy to :shipit:

**Pull request merged**